capstone/src/main/webapp/script.js

```diff
@@ -188,8 +190,65 @@ function setUpReplacements(text, lang) {
   submitButton.className = 'btn btn-primary';
   submitButton.onclick = function() { getReplacements(text, lang); };
   replacementsContainer.appendChild(submitButton);
+
+  // Set up extremes analysis button
+  const extremesButton = document.createElement('button');
+  extremesButton.innerHTML = 'Get Extremes (no input)';
+  extremesButton.className = 'btn btn-primary';
+  extremesButton.onclick = function() { getExtremes(text, lang); };
+  replacementsContainer.appendChild(extremesButton);
 }
- 
+
+/** Gets the most extreme word replacements on a string */
+async function getExtremes(text, lang) {
+  // Remove any previous output & set up loading spinner
+  const container = document.getElementById('perspective-datamuse-extremes');
+  container.innerHTML = '';
+  const loadingEl = document.createElement('p');
+  loadingEl.className = 'spinner-border';
+  container.appendChild(loadingEl);
+
+  // Get the words and all their possible replacements
+  const words = text.split(' ');
+  const mostToxic = { string : '', score : Number.MIN_VALUE};
+  const leastToxic = { string : '', score : Number.MAX_VALUE};
+  for (let word of words) {
+    const replacements = [];
+    for (let attribute of Object.values(DATAMUSE_ATTRIBUTES)) {
+      const datamuseResponse = await callDatamuse(word, attribute, 10);
```

Comment:
Although it might not have as  much an effect as it does when calling perspective, running these requests in parallel with `Promise.all` might be of use here.

Reply:
I did this optimization in "Optimize using parallel calls." You're right, unfortunately, it's not much faster